src/jobservice/pool/message_server.go

```diff
@@ -18,40 +19,75 @@ import (
 	"github.com/vmware/harbor/src/jobservice/utils"
 )
 
+const (
+	retryTimes = 5
+)
+
 //MessageServer implements the sub/pub mechanism via redis to do async message exchanging.
 type MessageServer struct {
-	context   context.Context
-	redisPool *redis.Pool
-	namespace string
-	callbacks map[string]reflect.Value //no need to sync
+	context      context.Context
+	redisPool    *redis.Pool
+	namespace    string
+	callbacks    map[string]reflect.Value //no need to sync
+	retryFlag    atomic.Value
```

Comment:
Need more context to understand why this has to be atomic value.

Reply:
Just to avoid -race problems. It assumes the start might be called by different goroutines. Actually it's only called once